newlib/libc/sys/vita/chown.c

```diff
@@ -0,0 +1,43 @@
+/*
+
+Copyright (C) 2022, vitasdk
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+
+*/
+#include <reent.h>
+
+#include <errno.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+int chown(const char *path, uid_t owner, gid_t group)
+{
+	struct _reent *reent = _REENT;
+	reent->_errno = ENOSYS;
```

Comment:
how about just `errno = ENOSYS`?
but error code could be `EINVAL`

Reply:
Yeah, you're right. I somehow assumed it should be reent.
error code is right, though, if we do nothing. But we can check gid/uid against 1 (which we return from getgid/getuid) and returt EINVAL or success. Nice catch